scripts/common/rook.sh

```diff
@@ -432,3 +432,42 @@ function add_rook_store_object_migration_status() {
     kubectl patch configmap kurl-migration-from-rook -n kurl --type merge -p '{"data":{"DID_MIGRATE_ROOK_OBJECT_STORE":"1"}}'
     export DID_MIGRATE_ROOK_OBJECT_STORE=1
 }
+
+# rook_maybe_migrate_from_openebs may migrate data from OpenEBS to Rook when all the following
+# conditions are met:
+# - Ekco version is >= 0.27.1.
+# - OpenEBS and Rook are selected on the Installer.
+# - Rook's minimum node count is set to a value > 1.
+# - The number of nodes on the cluster is >= than the Rook minimum node count.
+# - The 'scaling' storage class exists.
+function rook_maybe_migrate_from_openebs() {
+    semverCompare "$EKCO_VERSION" "0.27.1"
+    if [ "$SEMVER_COMPARE_RESULT" -lt "0" ]; then
+        return 0
+    fi
+    if [ -z "$ROOK_VERSION" ] || [ -z "$OPENEBS_VERSION" ]; then
+        return 0
+    fi
+    if [ -z "$ROOK_MINIMUM_NODE_COUNT" ] || [ "$ROOK_MINIMUM_NODE_COUNT" -le "1" ]; then
+        return 0
+    fi
+
+    if [ -z "$EKCO_AUTH_TOKEN" ]; then
+        logFail "Internal Error: an authentication token is required to start the OpenEBS to Rook multi-node migration."
+        return 0
+    fi
+
+    if [ -z "$EKCO_ADDRESS" ]; then
+        logFail "Internal Error: unable to determine network address of the kURL operator."
+        return 0
+    fi
+
+    # Initiate OpenEBS to Rook multi-node migration
+    if ! "${DIR}"/bin/kurl cluster migrate-multinode-storage --ekco-address "$EKCO_ADDRESS" --ekco-auth-token "$EKCO_AUTH_TOKEN" < /dev/tty; then
```

Comment:
you may want to wrap this in the prompts_can_prompt function. also if the prompt should default to no we have the flag yes that you should support.

Reply:
my question still stands here. what happens if there is no terminal? say this is invoked from automation.

Reply:
will do